app/src/main/java/com/example/drawingApp/DrawingField.kt

```diff
@@ -0,0 +1,96 @@
+package com.example.drawingApp
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.Path
+import android.util.AttributeSet
+import android.util.Log
+import android.view.MotionEvent
+import android.view.View
+import kotlin.math.abs
+
+
+@SuppressLint("ClickableViewAccessibility")
+class DrawingField @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context,attrs,defStyle) {
+    /*
+    what does this custom view do?
+    it should have an ability to draw inside its bounds
+    it should have a color to draw with
+        the color should be whatever the primary color is
+     */
+    //variables that should be available
+    /*
+    private paint variable
+    private bitMap variable
+
+     */
+    private var paint = Paint().apply{
+        color=Color.BLACK
+        style=Paint.Style.STROKE
+        strokeWidth=4f
+    //avoid the issue of dealing handling possible uninitialized values, black is expected on starting a drawing app after all
+    }
+    private lateinit var map:Bitmap
+    private var held = true
+    private  var path = Path()
+    private lateinit var calcCanvas:Canvas
+
+    init{
+        setOnTouchListener{_, mE ->
+            when(mE.action){
+                MotionEvent.ACTION_DOWN->{
+                    held = true
+                    path.moveTo(mE.x,mE.y)
+                }
+                MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL ->{
+                    held = false
+                }
+                MotionEvent.ACTION_MOVE ->{
+                    //can add an if statement that if the mE is out of bounds held = false
+                    if(held){
+                        path.lineTo(mE.x,mE.y)
+                        calcCanvas=Canvas(map)
+                        invalidate()
+                    }
+                }
+            }
+            true
+        }
+    }
+
+    //functions
+
+    fun updateColor(c:Int){
+        paint = Paint().apply {
+            color = c
+            style=Paint.Style.STROKE
+            strokeWidth=4f
+        }
+    }
+
+
+
+    override fun onDraw(canvas: Canvas?) {
+        super.onDraw(canvas)
+
+        canvas?.let { c ->
+            c.drawColor(Color.parseColor("#eeeeee"))
```

Comment:
Parsing a string color is slow, just cache the color. You're doing this each ondraw

Reply:
background is only drawn once now